Styles/Classes/UIElement+Extensions/UITextView+Styles.m

```diff
@@ -0,0 +1,119 @@
+//  Copyright © 2018 Inloop, s.r.o. All rights reserved.
+
+#import "UITextView+Styles.h"
+#import <objc/runtime.h>
+#import "Swizzle.h"
+#import "UIView+Styles.h"
+#import <Styles/Styles-Swift.h>
+
+@interface UITextView ()
+@property (nonatomic, strong) NSMutableDictionary<NSNumber *, TextStyle *> *textStyles;
+@property (nonatomic, strong) NSMutableDictionary<NSNumber *, LayerStyle *> *layerStyles;
+@end
+
+@implementation UITextView (Styles)
+
+- (NSMutableDictionary *)textStyles {
+    NSMutableDictionary *stored = objc_getAssociatedObject(self, @selector(textStyles));
+    if (!stored) {
+        stored = [NSMutableDictionary new];
+        [self setTextStyles:stored];
+    }
+    return stored;
+}
+
+- (NSMutableDictionary *)layerStyles {
+    NSMutableDictionary *stored = objc_getAssociatedObject(self, @selector(layerStyles));
+    if (!stored) {
+        stored = [NSMutableDictionary new];
+        [self setLayerStyles:stored];
+    }
+    return stored;
+}
+
+- (void)setLayerStyles:(NSMutableDictionary *)layerStyles {
+    objc_setAssociatedObject(self, @selector(layerStyles), layerStyles, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+
+- (void)setTextStyles:(NSMutableDictionary *)textStyles {
+    objc_setAssociatedObject(self, @selector(textStyles), textStyles, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+
+
++ (void)load {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        Class class = [self class];
+        swizzle_instance_method(class, @selector(becomeFirstResponder), @selector(swizzle_becomeFirstResponder));
+        swizzle_instance_method(class, @selector(resignFirstResponder), @selector(swizzle_resignFirstResponder));
+        swizzle_instance_method(class, @selector(awakeFromNib), @selector(swizzle_awakeFromNib));
+    });
+}
+
+- (BOOL)swizzle_becomeFirstResponder {
+    [self updateStylesForState:kEditing];
+    return [super becomeFirstResponder];
+}
+
+- (BOOL)swizzle_resignFirstResponder {
+    [self updateStylesForState:kInactive];
```

Comment:
[self swizzle_resignFirstResponder]

Don't forget that we are in category here. Not in subclass.